include/logging/log_output.h

```diff
@@ -104,6 +110,16 @@ static inline void log_output_ctx_set(const struct log_output *log_output,
 	log_output->control_block->ctx = ctx;
 }
 
+/** @brief Function for setting hostname of this device
+ *
+ * @param log_output	Pointer to the log output instance.
+ * @param hostname	Hostname of this device
+ */
+static inline void log_output_hostname_set(const struct log_output *log_output,
```

Comment:
Why do we need this? Isn't "hostname" a system-wide param (for which there should be generic accessors already)?

Reply:
There was `log_output_ctx_set()` so just following suit. Note that hostname support might not be compiled in so we need to handle that.